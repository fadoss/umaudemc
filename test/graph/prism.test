# Exporting PRISM files

umaudemc graph ../check/vending initial --passign uniform --format prism
umaudemc graph ../check/vending initial --passign uniform --format prism --slabel '{%t |= hasCake}'
umaudemc graph ../check/vending initial --passign uniform --format prism --aprops hasCake

umaudemc graph ../check/vending initial --passign 'uaction(cake=7, apple=5)' --format prism --slabel ''
umaudemc graph ../check/vending initial --passign "term(if A:Qid == 'cake then 2 else 3 fi)" --format prism

umaudemc graph ../pcheck/coin head --passign metadata --format prism --aprops head,tail
umaudemc graph ../pcheck/coin head 'choice(8 : ttail, 5 : thead) ; ttail' --passign strategy --format prism --aprops head,tail
umaudemc graph ../pcheck/coin head 'choice(8 : ttail, 5 : thead) *' --passign strategy --format prism --aprops head,tail

umaudemc graph ../pcheck/coin head 'choice(8 : ttail, 5 : thead) ; choice(3 : ttail, 7 : thead)' --passign strategy --format prism
