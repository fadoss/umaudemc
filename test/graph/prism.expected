dtmc

module M
	x : [0..7] init 0;

	// e e [empty]
	[] x=0 -> 1.0:(x'=1);
	// e [e]
	[] x=1 -> 0.5:(x'=2) + 0.5:(x'=6);
	// empty [e e]
	[] x=2 -> 0.5:(x'=3) + 0.5:(x'=5);
	// a [e]
	[] x=3 -> 1.0:(x'=4);
	// e a [empty]
	[] x=6 -> 1.0:(x'=3);

endmodule

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..7] init 0;

	// false
	[] x=0 -> 1.0:(x'=1);
	// false
	[] x=1 -> 0.5:(x'=2) + 0.5:(x'=6);
	// false
	[] x=2 -> 0.5:(x'=3) + 0.5:(x'=5);
	// false
	[] x=3 -> 1.0:(x'=4);
	// false
	[] x=6 -> 1.0:(x'=3);

endmodule

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..7] init 0;

	// e e [empty]
	[] x=0 -> 1.0:(x'=1);
	// e [e]
	[] x=1 -> 0.5:(x'=2) + 0.5:(x'=6);
	// empty [e e]
	[] x=2 -> 0.5:(x'=3) + 0.5:(x'=5);
	// a [e]
	[] x=3 -> 1.0:(x'=4);
	// e a [empty]
	[] x=6 -> 1.0:(x'=3);

endmodule
label "hasCake" = x=5;

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..7] init 0;

	// 
	[] x=0 -> 1.0:(x'=1);
	// 
	[] x=1 -> 0.16666666666666666:(x'=2) + 0.8333333333333334:(x'=6);
	// 
	[] x=2 -> 0.4166666666666667:(x'=3) + 0.5833333333333334:(x'=5);
	// 
	[] x=3 -> 1.0:(x'=4);
	// 
	[] x=6 -> 1.0:(x'=3);

endmodule

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..7] init 0;

	// e e [empty]
	[] x=0 -> 1.0:(x'=1);
	// e [e]
	[] x=1 -> 0.5:(x'=2) + 0.5:(x'=6);
	// empty [e e]
	[] x=2 -> 0.6:(x'=3) + 0.4:(x'=5);
	// a [e]
	[] x=3 -> 1.0:(x'=4);
	// e a [empty]
	[] x=6 -> 1.0:(x'=3);

endmodule

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..2] init 0;

	// head
	[] x=0 -> 0.6153846153846154:(x'=0) + 0.38461538461538464:(x'=1);
	// tail
	[] x=1 -> 0.6153846153846154:(x'=0) + 0.38461538461538464:(x'=1);

endmodule
label "head" = x=0;
label "tail" = x=1;

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..4] init 0;

	// head
	[] x=0 -> 0.6153846153846154:(x'=1) + 0.38461538461538464:(x'=2);
	// tail
	[] x=1 -> 1.0:(x'=3);
	// head
	[] x=2 -> 1.0:(x'=3);

endmodule
label "head" = x=0 | x=2;
label "tail" = x=1 | x=3;

rewards
	[] true: 1;
endrewards
mdp

module M
	x : [0..5] init 0;

	// head
	[] x=0 -> 1.0:(x'=1);
	// head
	[] x=0 -> 0.6153846153846154:(x'=2) + 0.38461538461538464:(x'=3);
	// tail
	[] x=2 -> 1.0:(x'=4);
	// tail
	[] x=2 -> 0.6153846153846154:(x'=2) + 0.38461538461538464:(x'=3);
	// head
	[] x=3 -> 1.0:(x'=1);
	// head
	[] x=3 -> 0.6153846153846154:(x'=2) + 0.38461538461538464:(x'=3);

endmodule
label "head" = x=0 | x=1 | x=3;
label "tail" = x=2 | x=4;

rewards
	[] true: 1;
endrewards
dtmc

module M
	x : [0..5] init 0;

	// head
	[] x=0 -> 0.6153846153846154:(x'=1) + 0.38461538461538464:(x'=2);
	// tail
	[] x=1 -> 0.3:(x'=3) + 0.7:(x'=4);
	// head
	[] x=2 -> 0.3:(x'=3) + 0.7:(x'=4);

endmodule

rewards
	[] true: 1;
endrewards
